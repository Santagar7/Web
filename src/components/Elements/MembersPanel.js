import {useParams} from 'react-router-dom';
import axios from "axios";
import React from 'react';
import {useState, useEffect} from 'react';
axios.defaults.withCredentials = true;

const url = 'http://localhost:7000/';

function MembersPanel({setActive}){
    const {index} = useParams();
    const [show,setShow] = useState(true);
    const [members,setMembers] = useState([]);
    
    useEffect(()=>{
        axios.get(url+`members/${index}`)
            .then(resp=>{
                setMembers(resp.data.members);
        })
    },[show]);

    function ShowMembersPanel(){
        return (
            <div className="center">
                <h1>All Members</h1>
                <div id="members_list" className="members">
                    {members.map(item=>(<h1 key={item}>{item}</h1>))}
                </div>
                <form>
                    <button type="button" onClick={()=>{setShow(!show)}}>Add member</button>
                </form>
                <form id="cancel">
                    <button type="button" className="exit" onClick={()=>{setActive(false)}}>Cancel</button>
                </form>
            </div>
        )
    }

    function AddMembersPanel(){
        return (
            <div className="center">
            <h1 id="message">Add member</h1>
            <form>
                <input id="email" type="email" required placeholder="Email"/>
                <button type="button" onClick={AddNewMember}>Submit</button>
            </form>
            <form id='cancel'>
                <button type="button" className="exit" onClick={()=>{setShow(!show)}}>Cancel</button>
            </form>
        </div>  
        )
    }

    function AddNewMember(e){
        e.preventDefault();
        const email = document.getElementById('email').value;
        let elem = document.getElementById('message');
        axios.post(url+`add_members/${index}`,{
            'email':email
        }).then((resp)=>{
            if(resp.data.message==='Success'){
                setShow(!show);
            }
            else{
                elem.textContent = resp.data.message;
                elem.style.color = 'crimson';
            }
        })
    }

    return (
        <div className='panel'>
            {show?<ShowMembersPanel/>:<AddMembersPanel/>}
        </div>)
}

export default MembersPanel;