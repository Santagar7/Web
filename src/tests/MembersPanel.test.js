import { BrowserRouter } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MembersPanel from '../components/Elements/MembersPanel';

test('should render correctly', () => {
    const tree = renderer.create(<BrowserRouter><MembersPanel/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

test('check create new action', () => {
    render(
        <BrowserRouter>
            <MembersPanel />
        </BrowserRouter>,
    );

    const linkElement = screen.getByText(/All Members/i);
    expect(linkElement).toBeInTheDocument();
});

test('check actions panel', () => {
    render(
        <BrowserRouter>
            <MembersPanel />
        </BrowserRouter>,
    );

    const linkElement = screen.getByText(/Cancel/i);
    expect(linkElement).toBeInTheDocument();

    const submit = screen.getByText(/Add member/i);
    fireEvent.click(submit);

});
