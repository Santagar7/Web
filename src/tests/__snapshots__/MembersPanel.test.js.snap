// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="panel"
>
  <div
    className="center"
  >
    <h1>
      All Members
    </h1>
    <div
      className="members"
      id="members_list"
    />
    <form>
      <button
        onClick={[Function]}
        type="button"
      >
        Add member
      </button>
    </form>
    <form
      id="cancel"
    >
      <button
        className="exit"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </form>
  </div>
</div>
`;
